### PR TITLE
[RFR] Ninja TLC

### PR DESCRIPTION
**Pull request by MadmanMartian**
Walls are now consistent with other sharpness flag checks, and require harm intent

humans being thrown towards the station are now encapsulated within an airbag so they don't fucking die on impact. Airbags now only dissipate if the user moves.

Ninja equipment dissipating now only informs the user, rather than using visible message

More katana suicide reasons for funsies

:cl:
 * tweak: Cutting a wall with an energy sword now requires the user to be in harm intent, to mirror the other structure cutting functions
 * tweak: Humans being thrown towards the station are now encapsulated within an airbag.
 * tweak: Airbags now release their content when the user inside moves, rather than lasting for strictly 5 seconds.
 * tweak: Ninja equipment dissipating now only informs the user, rather than being a visible message

### CODE REVIEW

**Comment by Kurfursten** [comment]
>Cutting a wall requires harm intent

Please no. When I weld a wall, I don't want to be on harm intent. Same with swording a wall.

**Comment by MadmanMartian** [comment]
But welders do their own thing, rather than having a flag for CUT_WALL.

Everywhere else, swording requires harm intent, it's really non-intuitive for them all to be different.

**Comment by Kurfursten** [comment]
It's unintuitive that elsewhere it does require the harm intent. If you don't want to cut the wall, don't click it with your sword and stand there while the UI bar fills up.

**Comment by SonixApache** [comment]
consistency good

good changes, good thinking

**Comment by MadmanMartian** [comment]
Don't merge, having a hal moment

**Comment by MadmanMartian** [comment]
>secret repo conflict

It was nice knowing you

**Comment by MadmanMartian** [comment]
Right, changes made in the last commit:

 - The powerglove (It's so bad) no longer just holds charge in a reservoir variable, but gives and draws from the detected cell of the user
 - The ninja suit now holds a cell, initially empty, but is charged over time. Now you can EMP a ninja.

 - I saw some devil-spawns, so nuked the entire antag joining process just to be sure. Now, rather than rulesets spawning to pull somebody from the arrivals shuttle, the role itself handles the pulling of people to different areas, so that person never makes it to the arrivals shuttle in the first place.

File: code/datums/gamemode/role/ninja.dm [hunk 1]
```diff
@@ -727,7 +761,7 @@ Suit and assorted
 	to_chat(user, "[message]")
 
 /obj/item/weapon/katana/hesfast/suicide_act(mob/user)
-	visible_message("<span class='danger'>[user] is slicing \his chest open with the [src.name]! It looks like \he's trying to commit sudoku.</span>")
+	visible_message("<span class='danger'>[user] is slicing \his chest open with the [src.name]! It looks like \he's trying to commit [pick("seppuku","sudoku","harikari","crossword puzzle")].</span>")
```
Comment (SonixApache) [inline code/datums/gamemode/role/ninja.dm:773]: This one specifically aims at the chest to commit sudoku, unlike the normal one which is aimed at the belly. Don't feel like this works.

File: code/modules/customitems/item_spawning.dm [hunk 2]
```diff
@@ -1,87 +1,111 @@
...
-			Item.forceMove(get_turf(M.loc))
-
-		HackProperties(Item,propadjust)
+			//skip:
+			if (ok == 0) // Finally, since everything else failed, place it on the ground
+	//			testing("Plopped onto the ground!")
+				Item.forceMove(get_turf(M.loc))
+
+			HackProperties(Item,propadjust)
+	to_chat(world, "Equip custom items, [M.disabilities]")
```
Comment (SonixApache) [inline code/modules/customitems/item_spawning.dm]: 😯

**Comment by SonixApache** [comment]
The devil spawn thing might be caused by the fact that I create the ninja's body in their equip proc, rather than just using the `generate_ruleset_body` proc.

**Comment by DamianX** [comment]
This is getting super-unatomic, what are you doing?

**Comment by tuypo1** [comment]
What the fuck ever happened to keeping things atomic.

**Comment by MadmanMartian** [comment]
I got carried away, I'll shift the role datum stuff to a different PR

**Comment by MadmanMartian** [comment]
There, back to being atomic

File: data/mode.txt [hunk 3]
```diff
@@ -1 +1 @@
-sandbox
+Dynamic Mode
```
Comment (jknpj) [inline data/mode.txt:1]: :thinking:

Comment (MadmanMartian) [reply data/mode.txt:1]: dangert, why didn't you let me fix this before you merged it?

**Pull request merged**